sklearn/cross_decomposition/_pls_.py

```diff
@@ -420,6 +420,35 @@ def transform(self, X, Y=None, copy=True):
 
         return x_scores
 
+    def inverse_transform(self, X):
+        """Transform data back to its original space.
+
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_components)
+            New data, where n_samples is the number of samples
+            and n_components is the number of pls components.
+        copy : bool, default=True
+            Whether to copy X, or perform in-place normalization.
+
+        Returns
+        -------
+        X_original array-like of shape (n_samples, n_features)
+
+        Notes
+        This transformation will only be exact if n_components=n_features
+        -----
+        """
+        check_is_fitted(self)
+        X = check_array(X, copy=False, dtype=FLOAT_DTYPES)
+        # From pls space to original space
+        X_original = np.matmul(X, self.x_loadings_.T)
```

Comment:
```suggestion
        np.matmul(X, self.x_loadings_.T, out=X)
```

this way you can do it in place.

You should also add a test for this case.

Reply:
So I came up with this:

```python
def inverse_transform(self, X, copy=True):
        check_is_fitted(self)
        X = check_array(X, copy=copy, dtype=FLOAT_DTYPES)
        # From pls space to original space
        np.matmul(X, self.x_loadings_.T, out=X)

        # Denormalize
        X *= self.x_std_
        X += self.x_mean_
        return X
```

And the extra test:
```python
    # Check inplace usage
    plsca.inverse_transform(Xr,copy=False)
    assert_array_almost_equal(Xr, X,
                              err_msg="inverse_transform failed")
```

I tested it and it works with both inplace. Just checking this is what you meant before commiting it :)

Reply:
yeah this looks good to me, I think.